api/pkg/service/resource/resource.go

```diff
@@ -346,6 +352,41 @@ func filterByKind(t string) func(db *gorm.DB) *gorm.DB {
 	}
 }
 
+func filterByKinds(t []string) func(db *gorm.DB) *gorm.DB {
+	if len(t) == 0 {
+		return noop
+	}
+
+	return func(db *gorm.DB) *gorm.DB {
+		t = lower(t)
+		return db.Where("LOWER(kind) IN (?)", t)
+	}
+}
+
+func filterByTags(t []uint) func(db *gorm.DB) *gorm.DB {
+	if t == nil {
+		return noop
+	}
+
+	return func(db *gorm.DB) *gorm.DB {
+
+		return db.Where("id IN (?)", t)
+	}
+}
+
+func queryResourcesByTags(t []string, db *gorm.DB) (resID []uint) {
```

Comment:
should we move this above `resourcesForQuery` and make this method of service and if the query fails return a `fetchError`?